packages/base/src/features/LegacyDateFormats.ts

```diff
@@ -0,0 +1,56 @@
+import { getFormatSettings } from "../InitialConfiguration.js";
+import type { FormatSettings } from "../config/FormatSettings.js";
+import { registerFeature } from "../FeaturesRegistry.js";
+
+// legacy ABAP date formats
+type LegacyDateFormat = "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9" | "A" | "B" | "C" | undefined;
+// Allows specifying the customizing data for Islamic calendar support
+// dateFormat - The date format // check LegacyDateFormat type for supported values
+// islamicMonthStart - The Islamic date in string format // 14360101
+// gregDate - Corresponding Gregorian date to the Islamic one in string format // 20141024
+type CalendarFormat = {
```

Comment:
I think we can name IslamicToGregorianMapping or something along these lines to describe it more precisely